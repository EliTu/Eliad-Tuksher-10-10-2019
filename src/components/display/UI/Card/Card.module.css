@import '../../../../variables.css';

.CardStyles {
    width: 20%;
    height: 15rem;
    margin: 1.5rem 0.8rem;
    padding: 1rem 2rem;
    border: var(--border-thick);
    border-color: var(--color-main-black-opaque);
    display: flex;
    flex-flow: column;
    align-content: center;
    justify-content: center;
    align-items: center;
}

.CardStyles h3 {
    font-size: var(--text-size-large);
}

.CardStyles h4 {
    font-size: var(--text-size-medium);
}

.CardStyles p {
    font-size: var(--text-size-medium);
}