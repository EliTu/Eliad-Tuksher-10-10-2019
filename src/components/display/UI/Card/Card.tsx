import React from 'react';
import styles from './Card.module.css';

interface IProps {
	mainHeading: string;
	date: string;
	link?: string;
	description: string;
	temperatures: string;
}

const Card: React.FC<IProps> = ({
	mainHeading,
	date,
	description,
	link,
	temperatures,
}) => {
	const {
		CardStyles,
		headingStyle,
		dateStyle,
		temperatureStyle,
		descriptionStyle,
	} = styles;

	return (
		<div className={CardStyles}>
			<h3 className={headingStyle}>
				<a href={link}>{mainHeading}</a>
			</h3>
			<h3 className={dateStyle}>{date}</h3>
			<h4 className={temperatureStyle}>{temperatures}</h4>
			<p className={descriptionStyle}>{description}</p>
		</div>
	);
};

export default Card;
