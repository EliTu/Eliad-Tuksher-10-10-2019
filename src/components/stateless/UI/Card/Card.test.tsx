import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import Card from './Card';

describe('Card component', () => {
	let component: ShallowWrapper;
	beforeEach(
		() =>
			(component = shallow(
				<Card
					mainHeading={'card-test'}
					secondaryHeading={'1'}
					info={'abc'}
				/>
			))
	);

	it('should render without errors', () => {
		expect(component).toMatchSnapshot();
		expect(component.length).toBe(1);
		expect(component.length).not.toBe(2);
		expect(component).toHaveClassName('CardStyles');
	});

	it('should render a div with an h3, h4 and p tags, containing the info passed as props', () => {
		expect(
			component
				.children()
				.find('h3')
				.text()
		).toBe('card-test');
		expect(
			component
				.children()
				.find('h4')
				.text()
		).toBe('1');
		expect(
			component
				.children()
				.find('p')
				.text()
		).toBe('abc');
	});
});
