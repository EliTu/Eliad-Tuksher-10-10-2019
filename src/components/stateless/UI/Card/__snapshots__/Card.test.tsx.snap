// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component should render without errors 1`] = `
<div
  className="CardStyles"
>
  <h3>
    card-test
  </h3>
  <h4>
    1
  </h4>
  <p>
    abc
  </p>
</div>
`;
